File: lec1/examples/linux-c/redirect.c

// redirect.c: run a command with output redirected
#include <stdio.h>
#include <unistd.h>
#include <fcntl.h>
#include <sys/types.h>
#include <sys/wait.h>
#include <stdlib.h>
int
main()
{
  int pid;

  pid = fork();
  if(pid == 0){
    close(1);
    open("output.txt", O_WRONLY|O_CREAT);

    char *argv[] = { "echo", "this", "is", "redirected", "echo", 0 };
    execv("echo", argv);
    printf("exec failed!\n");
    exit(1);
  } else {
    wait((int *) 0);
  }
}

File: lec1/examples/linux-rust/src/redirect.rs
use fork::{fork, Fork};
use exec::Command;
fn main() {
    let pid = fork();
    if pid == 0 {
        close(1);
        open("output.txt", O_WRONLY | O_CREAT);
        let argv = [ "echo", "this", "is", "redirected", "echo", 0 ];
        exec("echo", argv);
        println!("exec failed!\n");
        exit(1);
    } else {
        wait(0);
    }
}